src/main/java/com/squareup/javapoet/TypeName.java

```diff
@@ -160,16 +164,18 @@ public boolean isBoxedPrimitive() {
    */
   public TypeName box() {
     if (keyword == null) return this; // Doesn't need boxing.
-    if (this == VOID) return BOXED_VOID;
-    if (this == BOOLEAN) return BOXED_BOOLEAN;
-    if (this == BYTE) return BOXED_BYTE;
-    if (this == SHORT) return BOXED_SHORT;
-    if (this == INT) return BOXED_INT;
-    if (this == LONG) return BOXED_LONG;
-    if (this == CHAR) return BOXED_CHAR;
-    if (this == FLOAT) return BOXED_FLOAT;
-    if (this == DOUBLE) return BOXED_DOUBLE;
-    throw new AssertionError(keyword);
+    TypeName boxed = null;
+    if (keyword.equals(VOID.keyword)) boxed = BOXED_VOID;
+    if (keyword.equals(BOOLEAN.keyword)) boxed = BOXED_BOOLEAN;
+    if (keyword.equals(BYTE.keyword)) boxed = BOXED_BYTE;
+    if (keyword.equals(SHORT.keyword)) boxed = BOXED_SHORT;
+    if (keyword.equals(INT.keyword)) boxed = BOXED_INT;
+    if (keyword.equals(LONG.keyword)) boxed = BOXED_LONG;
+    if (keyword.equals(CHAR.keyword)) boxed = BOXED_CHAR;
+    if (keyword.equals(FLOAT.keyword)) boxed = BOXED_FLOAT;
+    if (keyword.equals(DOUBLE.keyword)) boxed = BOXED_DOUBLE;
+    if (boxed == null) throw new AssertionError("cannot box " + keyword);
```

Comment:
If we match we should short-circuit rather than needlessly testing conditionals we know will fail.
```suggestion
    if (keyword.equals(VOID.keyword)) boxed = BOXED_VOID;
    else if (keyword.equals(BOOLEAN.keyword)) boxed = BOXED_BOOLEAN;
    else if (keyword.equals(BYTE.keyword)) boxed = BOXED_BYTE;
    else if (keyword.equals(SHORT.keyword)) boxed = BOXED_SHORT;
    else if (keyword.equals(INT.keyword)) boxed = BOXED_INT;
    else if (keyword.equals(LONG.keyword)) boxed = BOXED_LONG;
    else if (keyword.equals(CHAR.keyword)) boxed = BOXED_CHAR;
    else if (keyword.equals(FLOAT.keyword)) boxed = BOXED_FLOAT;
    else if (keyword.equals(DOUBLE.keyword)) boxed = BOXED_DOUBLE;
    else throw new AssertionError(keyword);
```